src/actions/actionLineEditor.ts

```diff
@@ -0,0 +1,73 @@
+import { getNonDeletedElements } from "../element";
+import { LinearElementEditor } from "../element/linearElementEditor";
+import { isLinearElement } from "../element/typeChecks";
+import { ExcalidrawLinearElement } from "../element/types";
+import { getSelectedElements } from "../scene";
+import { register } from "./register";
+
+export const actionEnterLineEditor = register({
+  name: "enterLineEditor",
+  trackEvent: {
+    category: "canvas",
+    predicate: (appState) => !appState.zenModeEnabled,
+  },
+  contextItemPredicate: (elements, appState) => {
+    const selectedElements = getSelectedElements(elements, appState);
+    if (
+      selectedElements.length === 1 &&
+      isLinearElement(selectedElements[0]) &&
+      appState.editingLinearElement?.elementId !== selectedElements[0].id
+    ) {
+      return true;
+    }
+    return false;
+  },
+  perform(elements, appState, _, app) {
+    const selectedElement = getSelectedElements(
+      getNonDeletedElements(elements),
+      appState,
+      true,
+    )[0] as ExcalidrawLinearElement;
+    return {
+      appState: {
+        ...appState,
+        editingLinearElement: new LinearElementEditor(
+          selectedElement,
+          app.scene,
+        ),
+      },
+      commitToHistory: false,
+    };
+  },
+  contextItemLabel: "labels.lineEditor.edit",
+});
+
+export const actionExitLineEditor = register({
+  name: "exitLineEditor",
+  trackEvent: {
+    category: "canvas",
+    predicate: (appState) => !appState.zenModeEnabled,
+  },
```

Comment:
probably not useful to track exiting

Reply:
I think its ok to keep in case we want to ever track how user exits (eg keyboard vs context menu)

Reply:
problem is we're not tracking (and grouping) `esc`, and `cmd-enter` shouldn't work for exiting either.

Reply:
Now that actions are combined, do you want to explicitly not track when exiting? I am not sure why we are talking about "cmd+enter" here but "Esc" and "Exit line editor" via context menu is what I was pointing to. But in any case I do agree its not that imp data so its ok not to track